scripts/sync-interface-data.js

```diff
@@ -0,0 +1,385 @@
+      throw new Error('Destroying actions is not yet supported');
+    }
+
+    this.logger.info(`${actionsToCreate.length} actions to create`);
+    this.logger.info(`${actionsToUpdate.length} actions to update`);
+    this.logger.info(`${actionsToDestroy.length} actions to destroy`);
+
+    // Perform all insertions and updates.
+    await Promise.all([
+      ...actionsToCreate.map(actionData => createRecord(trx, 'actions', getActionTargetData, actionData, data.themeIdMap)),
+      ...actionsToUpdate.map(async actionData => {
+        const sourceAction = data.source.actions.find(action => action.id === actionData.origin_id);
+        await updateRecord(trx, 'actions', getActionTargetData, sourceAction, data.themeIdMap);
+      })
+    ]);
+  }
+
+  /**
+   * Synchronizes themes in the target database from the source database.
+   *
+   * A `themeIdMap` property will be added to the `data` argument.
+   * This will be an object with source theme IDs as keys and the corresponding target theme IDs as values.
+   *
+   * @param {Transaction} trx - Knex transaction.
+   * @param {Object} data - Data from the source and target database.
+   * @param {Object} data.source - Data from the source database (see `loadSourceData`).
+   * @param {Object} data.target - Data from the target database (see `loadTargetData`).
+   */
+  async synchronizeThemes(trx, data) {
+
+    const sourceThemeIds = map(data.source.themes, 'id');
+    const targetThemeOriginIds = map(data.target.themes, 'origin_id');
+
+    const themesToCreate = data.source.themes.filter(theme => !includes(targetThemeOriginIds, theme.id));
+    const themesToUpdate = data.target.themes.filter(theme => includes(sourceThemeIds, theme.origin_id));
+    const themesToDestroy = data.target.themes.filter(theme => !includes(sourceThemeIds, theme.origin_id));
+    if (themesToDestroy.length) {
+      throw new Error('Destroying themes is not yet supported');
+    }
+
+    this.logger.info(`${themesToCreate.length} themes to create`);
+    this.logger.info(`${themesToUpdate.length} themes to update`);
+    this.logger.info(`${themesToDestroy.length} themes to destroy`);
+
+    // Perform all insertions and updates.
+    const result = await Promise.all([
+      ...themesToCreate.map(themeData => createRecord(trx, 'themes', getThemeTargetData, themeData)),
+      ...themesToUpdate.map(async themeData => {
+        const sourceTheme = data.source.themes.find(theme => theme.id === themeData.origin_id);
+        await updateRecord(trx, 'themes', getThemeTargetData, sourceTheme);
+      })
+    ]);
+
+    const createdThemeIds = result.slice(0, themesToCreate.length);
+
+    // Build a theme ID map with source theme IDs as keys and the corresponding target theme IDs as values.
+    data.themeIdMap = {
+      ...themesToCreate.reduce((memo, theme, i) => ({ ...memo, [theme.id]: createdThemeIds[i] }), {}),
+      ...themesToUpdate.reduce((memo, theme) => ({ ...memo, [theme.origin_id]: theme.id }), {})
```

Comment:
Maybe rename the `theme` parameter to better stress that it's not always the "same". For `themesToCreate.reduce`, `theme` represents a "source" theme while it represents a "target" theme for `themesToUpdate.reduce`.

Reply:
:+1: